src/Components/Aspire.Microsoft.EntityFrameworkCore.SqlServer/AspireSqlServerEFCoreSqlClientExtensions.cs

```diff
@@ -39,6 +40,29 @@ public static class AspireSqlServerEFCoreSqlClientExtensions
         string connectionName,
         Action<MicrosoftEntityFrameworkCoreSqlServerSettings>? configureSettings = null,
         Action<DbContextOptionsBuilder>? configureDbContextOptions = null) where TContext : DbContext
+    {
+        builder.AddSqlServerDbContext<TContext>(connectionName, configureSettings, configureDbContextOptions, null);
+    }
+
+    /// <summary>
+    /// Registers the given <see cref="DbContext" /> as a service in the services provided by the <paramref name="builder"/>.
+    /// Enables db context pooling, retries, health check, logging and telemetry for the <see cref="DbContext" />.
+    /// </summary>
+    /// <typeparam name="TContext">The <see cref="DbContext" /> that needs to be registered.</typeparam>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="connectionName">A name used to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    /// <param name="configureSettings">An optional delegate that can be used for customizing options. It's invoked after the settings are read from the configuration.</param>
+    /// <param name="configureDbContextOptions">An optional delegate to configure the <see cref="DbContextOptions"/> for the context.</param>
+    /// <param name="sqlServerOptionsAction">An optional delegate to configure the <see cref="SqlServerDbContextOptionsBuilder"/> for the context.</param>
+    /// <remarks>Reads the configuration from "Aspire:Microsoft:EntityFrameworkCore:SqlServer:{typeof(TContext).Name}" config section, or "Aspire:Microsoft:EntityFrameworkCore:SqlServer" if former does not exist.</remarks>
+    /// <exception cref="ArgumentNullException">Thrown if mandatory <paramref name="builder"/> is null.</exception>
+    /// <exception cref="InvalidOperationException">Thrown when mandatory <see cref="MicrosoftEntityFrameworkCoreSqlServerSettings.ConnectionString"/> is not provided.</exception>
+    public static void AddSqlServerDbContext<[DynamicallyAccessedMembers(RequiredByEF)] TContext>(
+        this IHostApplicationBuilder builder,
+        string connectionName,
+        Action<MicrosoftEntityFrameworkCoreSqlServerSettings>? configureSettings = null,
+        Action<DbContextOptionsBuilder>? configureDbContextOptions = null,
+        Action<SqlServerDbContextOptionsBuilder>? sqlServerOptionsAction = null) where TContext : DbContext
```

Comment:
This should be required here.
```suggestion
        Action<SqlServerDbContextOptionsBuilder> sqlServerOptionsAction) where TContext : DbContext
```